src/php/bin/run_tests.sh

```diff
@@ -24,3 +24,9 @@ source ./determine_extension_dir.sh
 export DYLD_LIBRARY_PATH=$root/libs/$config
 php $extension_dir -d max_execution_time=300 $(which phpunit) -v --debug \
   ../tests/unit_tests
+
+export ZEND_DONT_UNLOAD_MODULES=1
+export USE_ZEND_ALLOC=0
+valgrind --error-exitcode=10 --leak-check=yes php $extension_dir -d max_execution_time=300 \
```

Comment:
how long does this take to execute?

To be honest I'm not a fan of  piggybacking on an existing test suite like this. Ideally, this would be a separate test suite that would clearly indicate it's php with valgrind (e.g. we have separate sanitizers for C/C++) and we'd have the ability to run in separately if we needed to (e.g. because it might take long to execute or if we just wanted to reorganize tests if we wanted to). Also, running 2 test suite one by one in a shell script has some disadvantages  - e.g. it's hard to navigate the test results (and it's easy to miss things) and also the second suite won't run at all if the first one fails. So I'm somewhat against this (but not strongly).
 
If you believe there's a big value in the addition and you're willing to take the burden of maintenance (once we restruture the tests it might be harder to run stuff in this fashion) - or risk removing the suite if it become imcompatible with the way we run tests in the future, I'm probably ok with this.

Reply:
Thank you for the help!
It take about 4s-5s to finish. My idea is that if the unit tests fail, valgrind will report wrong leak information, so it is better not to run. I can add a flag for this script to choose which one to run.

Valgrind is just a short term solution. I am planning to use asan, tsan from clang the same as current infrastructure in C-core in the final.
I am trying to figure out asan tests in gRPC repo, can you offer me some suggestions? Like script to tests with asan locally? I am referring PR running asan for ruby: https://github.com/grpc/grpc/pull/11580

I can delete valgrind and change it to asan in this PR directly so there is no need to change these two later.

Reply:
It may also take some time before I figure out how to insert asan into php.
Currently, asan will make the program exit during grpc_init() and the error is:
*** Error in `php': double free or corruption (out): 0x0000603000052240 ***
I am not sure do I need some FLAG during compile grpc or not.
because loading protobuf.so doesn't have this error.

Reply:
ok, if it's that quick to build & run, it might be ok (and I fulfilled my duty about warning against this degenerating into something we don't want).

Please fix other comments first, then I'll take another look.

Reply:
Thanks for helping me to review it.

It has 113 tests and it can finish running by the time I press 'Enter' when there are no valgrind.

I am still trying to combine asan with php extension. Although I failed today, I'll try a little further. If asan works, I can remove valgrind from this PR and do the test like ruby's PR. If it is hard, I'll clean this PR, find a suitable place to run valgrind.